src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -513,6 +508,19 @@ private ImmutableArray<NamedTypeSymbol> MakeDeclaredInterfaces()
             }
         }
 
+        private static NamedTypeSymbol DecodeNullableAndTupleTypes(NamedTypeSymbol type, EntityHandle handle, PEModuleSymbol moduleSymbol)
+        {
+            // PROTOTYPE(NullableReferenceTypes): Should call NullableTypeDecoder.TransformOrEraseNullability
+            // here (see StaticNullChecking.UnannotatedAssemblies_09) but TransformOrEraseNullability results in
+            // a StackOverflowException when building Roslyn.sln.
+            if (moduleSymbol.UtilizesNullableReferenceTypes)
+            {
+                type = (NamedTypeSymbol)NullableTypeDecoder.TransformType(TypeSymbolWithAnnotations.Create(type), handle, moduleSymbol).TypeSymbol;
+            }
+            // PROTOTYPE(NullableReferenceTypes): Shouldn't be necessary to wrap and unwrap TypeSymbol.
+            return (NamedTypeSymbol)TupleTypeDecoder.DecodeTupleTypesIfApplicable(TypeSymbolWithAnnotations.Create(type, isNullableIfReferenceType: null), handle, moduleSymbol).TypeSymbol;
```

Comment:
>TypeSymbolWithAnnotations.Create(type, isNullableIfReferenceType: null) [](start = 82, length = 71)

Consider factoring the wrapping and unwrapping out, and leaving a comment.

```C#
var wrapped = TypeSymbolWithAnnotations.Create(type);
if (moduleSymbol.UtilizesNullableReferenceTypes)
    wrapped = ....;
return .... .TypeSymbol;
```

Reply:
Moved to a dedicated method.

---
In reply to: [194199107](https://github.com/dotnet/roslyn/pull/27589#discussion_r194199107) [](ancestors = 194199107)